bugtool/cmd/configuration.go

```diff
@@ -0,0 +1,247 @@
+	}
+	defer f.Close()
+
+	data, err := json.MarshalIndent(c, "", "\t")
+	if err != nil {
+		return fmt.Errorf("Cannot marshal config %s", err)
+	}
+	err = ioutil.WriteFile(path, data, 0644)
+	if err != nil {
+		return fmt.Errorf("Cannot write config %s", err)
+	}
+	return nil
+}
+
+func loadConfigFile(path string) (*BugtoolConfiguration, error) {
+	var content []byte
+	var err error
+	logrus.WithField(logfields.Path, path).Debug("Loading file")
+
+	content, err = ioutil.ReadFile(path)
+
+	if err != nil {
+		return nil, err
+	}
+
+	var c BugtoolConfiguration
+	err = json.Unmarshal(content, &c)
+	return &c, err
+}
+
+func catCommands() []string {
+	files := []string{
+		"/proc/sys/net/core/bpf_jit_enable",
+		"/proc/kallsyms",
+		"/etc/resolv.conf",
+		"/var/log/upstart/docker.log",
+		"/var/log/docker.log",
+		"/var/log/daemon.log",
+		"/var/log/messages",
+	}
+	// Only print the files that do exist to reduce number of errors in
+	// archive
+	commands := []string{}
+	for _, f := range files {
+		if _, err := os.Stat(f); os.IsNotExist(err) {
+			continue
+		}
+		commands = append(commands, fmt.Sprintf("cat %s", f))
+	}
+	// TODO: handle K8s case as well.
+	return commands
+}
+
+func copyConfigCommands(confDir string, k8sPods []string) []string {
+	commands := []string{}
+	type Location struct {
+		Src string
+		Dst string
+	}
+
+	locations := []Location{
+		{"/proc/config", fmt.Sprintf("%s/kernel-config", confDir)},
+		{"/proc/config.gz", fmt.Sprintf("%s/kernel-config.gz", confDir)},
+	}
+
+	if len(k8sPods) == 0 {
+		kernel, _ := execCommand("uname", "-r")
+		kernel = strings.TrimSpace(kernel)
+		l := Location{fmt.Sprintf("/boot/config-%s", kernel),
+			fmt.Sprintf("%s/kernel-config-%s", confDir, kernel)}
+		locations = append(locations, l)
+
+		for _, location := range locations {
+			if _, err := os.Stat(location.Src); os.IsNotExist(err) {
+				continue
+			}
+			commands = append(commands, fmt.Sprintf("cp %s %s", location.Src, location.Dst))
+		}
+	} else {
+		for _, pod := range k8sPods {
+			prompt := podPrefix(pod, "uname -r")
+			cmd, args := split(prompt)
+			kernel, _ := execCommand(cmd, args...)
+			kernel = strings.TrimSpace(kernel)
+			l := Location{fmt.Sprintf("/boot/config-%s", kernel),
+				fmt.Sprintf("%s/kernel-config-%s", confDir, kernel)}
+			locations = append(locations, l)
+
+			for _, location := range locations {
+				kubectlArg := fmt.Sprintf("%s/%s:%s", k8sNamespace, pod, location.Src)
+				cmd := fmt.Sprintf("%s %s %s %s", "kubectl", "cp", kubectlArg, location.Dst)
+				commands = append(commands, cmd)
+			}
+		}
+	}
+	return commands
+}
+
+func copyCiliumInfoCommands(ciliumDir string, k8sPods []string) []string {
+	// Most of the output should come via debuginfo but also adding
+	// these ones for skimming purposes
+	ciliumCommands := []string{
+		"cilium debuginfo",
+		"cilium config",
+		"cilium bpf tunnel list",
+		"cilium bpf lb list",
+		"cilium bpf endpoint list",
+		"cilium bpf ct list global",
```

Comment:
be aware that we can also have `ct list epID` if ct is local

Reply:
This should already be covered by debuginfo https://github.com/cilium/cilium/blob/master/cilium/cmd/debuginfo.go#L86